python/test/unit/hopper/test_experimental_tma.py

```diff
@@ -230,3 +232,98 @@ def test_device_tensormap1d(dtype_str):
     # Check results are correct
     torch.testing.assert_close(unwrap_tensor(inp), unwrap_tensor(out))
     torch.testing.assert_close(unwrap_tensor(inp), unwrap_tensor(inp_copy))
+
+
+@requires_tma
+@pytest.mark.parametrize("dtype_str", tma_dtypes)
+def test_tensor_descriptor_load(dtype_str):
+
+    @triton.jit
+    def kernel(out_ptr, a_ptr, M, N, M_BLOCK: tl.constexpr, N_BLOCK: tl.constexpr):
+        desc = tl._experimental_make_tensor_descriptor(
+            a_ptr,
+            shape=[M, N],
+            strides=[N, 1],
+            block_shape=[M_BLOCK, N_BLOCK],
+        )
+
+        assert desc.shape[0] == M
+        assert desc.shape[1] == N
+        assert desc.strides[0] == N
+        assert desc.strides[1] == 1
+        assert desc.block_shape == [M_BLOCK, N_BLOCK]
+        block = desc.load([M_BLOCK, 2 * N_BLOCK])
+        idx = tl.arange(0, M_BLOCK)[:, None] * N_BLOCK + tl.arange(0, N_BLOCK)[None, :]
+        tl.store(out_ptr + idx, block)
+
+    def alloc_fn(size: int, align: int, stream: Optional[int]):
```

Comment:
So the user must have the knowledge to figure out the size of the scratch buffer. Can you justify how it is compatible with our Autotuner?

Reply:
Not sure I understand your question. The user provided allocation function is given the size by the launcher code. During autotuning the allocation function will be called multiple times with different sizes to allocate new memory.

Reply:
My question is more about how we implement the `alloc_fn` when using the autotuner. The block sizes are changed based on the configurations passed by `triton.Config`.

Reply:
The size passed to the allocation function is the total allocation size, so the allocation function doesn't need to know anything about the config.

See for example in the next test down where there is a non-zero grid and so I assert:
```python
        assert size == 128 * (grid_m * grid_n)
```

Reply:
Just to make sure I understand it clearly.

If the maximum number of grids is 100x100, then we will allocate 100x100x128 anyway for all configurations

Reply:
No, say you have one config is 100x100 and another is 200x100 then you the allocation function will be called several times with `size = 100*100*128` and several times with `size = 200*100*128`.

Reply:
I see. Makes sense